src/legacy/core_plugins/metrics/public/setup.ts

```diff
@@ -19,9 +19,9 @@
 
 // @ts-ignore
 import { functionsRegistry } from 'plugins/interpreter/registries';
-import { CoreSetup, PluginsServiceSetup } from 'src/core/server';
+import { CoreSetup } from 'src/core/public';
 import { visualizations } from '../../visualizations/public';
-import { plugin } from './plugin';
+import { plugin, TsvbSetupPlugins } from './plugin';
```

Comment:
we should be exporting the `plugin` function from a separate `index.ts` file (this is the new platform convention: an `index.ts` that exports a function named `plugin`, and a `plugin.ts` that exports the plugin definition class)

Reply:
It contravenes the https://github.com/elastic/kibana/pull/39169#discussion_r297890809

Reply:
I've made the `public/index.ts` as an entry point of the `hacks`. `Plugin` and its `setup` config are stored in separate files.